File: typescript-client-sdk/src/sdk/models/shared/codatassessdatacontractsaccountcategoriespatchaccountcategorymodel.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { CodatAssessDataContractsAccountCategoriesConfirmedAccountCategoryModel } from "./codatassessdatacontractsaccountcategoriesconfirmedaccountcategorymodel";
import { CodatAssessDataContractsAccountCategoriesPatchAccountRefModel } from "./codatassessdatacontractsaccountcategoriespatchaccountrefmodel";
import { Expose, Type } from "class-transformer";


export class CodatAssessDataContractsAccountCategoriesPatchAccountCategoryModel extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "accountRef" })
  @Type(() => CodatAssessDataContractsAccountCategoriesPatchAccountRefModel)
  accountRef?: CodatAssessDataContractsAccountCategoriesPatchAccountRefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "confirmed" })
  @Type(() => CodatAssessDataContractsAccountCategoriesConfirmedAccountCategoryModel)
  confirmed?: CodatAssessDataContractsAccountCategoriesConfirmedAccountCategoryModel;
}

File: python-client-sdk/src/sdk/models/shared/codat_assess_datacontracts_accountcategories_patchaccountcategorymodel.py
from __future__ import annotations
import dataclasses
from ..shared import codat_assess_datacontracts_accountcategories_confirmedaccountcategorymodel as shared_codat_assess_datacontracts_accountcategories_confirmedaccountcategorymodel
from ..shared import codat_assess_datacontracts_accountcategories_patchaccountrefmodel as shared_codat_assess_datacontracts_accountcategories_patchaccountrefmodel
from dataclasses_json import Undefined, dataclass_json
from sdk import utils
from typing import Optional


@dataclass_json(undefined=Undefined.EXCLUDE)
@dataclasses.dataclass
class CodatAssessDataContractsAccountCategoriesPatchAccountCategoryModel:
    account_ref: Optional[shared_codat_assess_datacontracts_accountcategories_patchaccountrefmodel.CodatAssessDataContractsAccountCategoriesPatchAccountRefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('accountRef'), 'exclude': lambda f: f is None }})
    confirmed: Optional[shared_codat_assess_datacontracts_accountcategories_confirmedaccountcategorymodel.CodatAssessDataContractsAccountCategoriesConfirmedAccountCategoryModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('confirmed'), 'exclude': lambda f: f is None }})
    